mlir/lib/Analysis/AffineStructures.cpp

```diff
@@ -1035,6 +1036,20 @@ bool FlatAffineConstraints::isEmptyByGCDTest() const {
   return false;
 }
 
+// First, try the GCD test heuristic. If that doesn't find the set empty,
+// use the complete emptiness check provided by findIntegerSample, which gives
+// a definitive answer.
+bool FlatAffineConstraints::isIntegerEmpty() const {
+  if (isEmptyByGCDTest())
+    return true;
+  return !findIntegerSample().hasValue();
+}
+
+llvm::Optional<std::vector<int64_t>> FlatAffineConstraints::findIntegerSample() const {
```

Comment:
MLIR normally uses `SmallVector` instead of `std::vector`. Perhaps this should be changed.